packages/block-editor/src/components/block-toolbar/style.scss

```diff
@@ -105,7 +115,10 @@
 			position: absolute;
 			font-size: 16px;
 			right: 0;
-			bottom: $grid-unit-20;
+			height: $grid-unit-60;
+			top: 0;
+			display: inline-flex;
+			align-items: center;
```

Comment:
Since this dot is a font, its height may change depending on the `font-family` and `line-height`. So I explicitly set it to the same height as the button and aligned it vertically and centered.